rskj-core/src/main/java/co/rsk/core/bc/BlockExecutor.java

```diff
@@ -549,20 +558,65 @@ private BlockResult executeParallel(
                 block,
                 new LinkedList<>(executedTransactions.values()),
                 new LinkedList<>(receipts.values()),
-                new short[0],
-                totalGasUsed.longValue(),
-                Coin.valueOf(totalPaidFees.longValue()),
+                block.getHeader().getTxExecutionSublistsEdges(),
+                totalGasUsed,
+                totalPaidFees,
                 vmTrace ? null : track.getTrie()
         );
         profiler.stop(metric);
         logger.trace("End executeParallel.");
         return result;
     }
 
-    private void addFeesToRemasc(LongAccumulator remascFees, Repository track) {
-        long fee = remascFees.get();
-        if (fee > 0) {
-            track.addBalance(PrecompiledContracts.REMASC_ADDR, Coin.valueOf(fee));
+    private boolean detectCollisions(ReadWrittenKeysTracker tracker) {
+        Map<Long, Set<ByteArrayWrapper>> readKeysByThread = tracker.getReadKeysByThread();
+        Map<Long, Set<ByteArrayWrapper>> writtenKeysByThread = tracker.getWrittenKeysByThread();
+
+        Set<ByteArrayWrapper> readKeys = new HashSet<>();
+        Set<ByteArrayWrapper> writtenKeys = new HashSet<>();
+        int readKeysSize = 0;
+        int writtenKeysSize = 0;
+
+        Set<Long> threads = new HashSet<>();
+        threads.addAll(readKeysByThread.keySet());
+        threads.addAll(writtenKeysByThread.keySet());
+
+        for (Long threadId : threads) {
+
+            Set<ByteArrayWrapper> temporalReadKeys = readKeysByThread.getOrDefault(threadId, new HashSet<>());
+            Set<ByteArrayWrapper> temporalWrittenKeys = writtenKeysByThread.getOrDefault(threadId, new HashSet<>());
+
+            if (readKeysSize == 0 && writtenKeysSize == 0) {
+                readKeys = temporalReadKeys;
+                readKeysSize = temporalReadKeys.size();
+                writtenKeys = temporalWrittenKeys;
+                writtenKeysSize = temporalWrittenKeys.size();
+                continue;
+            }
+
+            writtenKeys.removeAll(temporalReadKeys);
+            writtenKeys.removeAll(temporalWrittenKeys);
```

Comment:
Use `Collections.disjoint` better and you can use two nested loops to compare all threads 1v1